forem/forem
# Migrate dashboard initializers to packs

## What type of PR is this? (check all applicable)


## Description
Per [#19002](https://github.com/forem/forem/issues/19002), we would like to migrate away from using the asset pipeline to serve our js initializers, in favor of webpacker. 

This issue is specifically concerned with the Dashboard Sort initializer, which attaches event listeners to the select dropdowns in the user's Dashboard view.

Since the dropdowns use InstantClick for navigating between options in the selects, I moved the initializer calls into javascript > packs > base.jsx, which currently uses `InstantClick.on('change')` to reinitialize video and podcast playback.

## Related Tickets & Documents



- Related Issue #19002
- Closes #19002

## QA Instructions, Screenshots, Recordings

Tested in Safari and Chrome at laptop and mobile widths. mobile_nav_dashboard select is expanded on the left margin as a "table of contents" in the wider view and appears as a screen-wide select dropdown at the top of the narrower "mobile" view.

Used admin user as test account. Created two posts by the user themselves and added a couple of comments to the older of the two, which allowed me to see the reordering when switching from `dashboard_sort`'s Recently Created to Most Comments option.

Checked that multiple re-selections would continue to update the DOM as well as append the correct `?sort=:param` to the URL up in the browser bar.

For `dashboard_author` I used the local app's admin panel to add the admin user to an organization and then used the top dropdown in the Create Post view to select that organization as I created a third post under the user. This allowed me to see the author select dropdown and switch between the Personal and [Organization Name] options. Verified the URL was updated in this case as well.

### UI accessibility concerns?

N/A, functionality seems to be identical to before. The initializer just lives in a different place and is tested now.

## Added/updated tests?

      have not been included_

Added jest coverage to ensure that a DOM containing the relevant select dropdowns will have the correct listeners attached when the initializer runs. One test per listener, specific to the ID of the listened-to select element.

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![Larry David saying pretty pretty good](https://media.giphy.com/media/m4jaaIGsn5KVO/giphy.gif)
